src/benchmarklib/tpch/tpch_query_generator.cpp

```diff
@@ -52,6 +70,339 @@ std::string TPCHQueryGenerator::get_preparation_queries() const {
+  //
+  // uniform_int_distributions should not be modified when they are used, but because we have no explicit thread safety
+  // guarantee, we make this thread-local, too.
+
+  static thread_local std::uniform_int_distribution<> material_dist{0, static_cast<int>(materials.size() - 1)};
+  static thread_local std::uniform_int_distribution<> region_dist{0, regions.count - 1};
+  static thread_local std::uniform_int_distribution<> segment_dist{0, c_mseg_set.count - 1};
+  static thread_local std::uniform_int_distribution<> nation_dist{0, nations.count - 1};
+  static thread_local std::uniform_int_distribution<> type_dist{0, p_types_set.count - 1};
+  static thread_local std::uniform_int_distribution<> color_dist{0, colors.count - 1};
+  static thread_local std::uniform_int_distribution<> shipmode_dist{0, l_smode_set.count - 1};
+  static thread_local std::uniform_int_distribution<> brand_char_dist{1, 5};
+  static thread_local std::uniform_int_distribution<> container_dist{0, p_cntr_set.count - 1};
+
+  // Will be filled with the parameters for this query and passed to the next method which builds the query string
+  std::vector<std::string> parameters;
+
+  switch (query_id) {
+    // Writing `1-1` to make people aware that this is zero-indexed while TPC-H query names are not
+    case 1 - 1: {
+      static thread_local std::uniform_int_distribution<> date_diff_dist{60, 120};
+      const auto date = calculate_date(boost::gregorian::date{1998, 12, 01}, 0, -date_diff_dist(random_engine));
+
+      parameters.emplace_back("'"s + date + "'");
+      break;
+    }
+
+    case 2 - 1: {
+      static thread_local std::uniform_int_distribution<> size_dist{1, 50};
+      const auto size = size_dist(random_engine);
+      const auto material = materials[material_dist(random_engine)];
+      const auto region = regions.list[region_dist(random_engine)].text;
+
+      parameters.emplace_back(std::to_string(size));
+      parameters.emplace_back("'%"s + material + "'");
+      parameters.emplace_back("'"s + region + "'");
+      parameters.emplace_back("'"s + region + "'");
+      break;
+    }
+
+    case 3 - 1: {
+      const auto segment = c_mseg_set.list[segment_dist(random_engine)].text;
+      static thread_local std::uniform_int_distribution<> date_diff_dist{0, 30};
+      const auto date = calculate_date(boost::gregorian::date{1995, 03, 01}, 0, date_diff_dist(random_engine));
+
+      parameters.emplace_back("'"s + segment + "'");
+      parameters.emplace_back("'"s + date + "'");
+      parameters.emplace_back("'"s + date + "'");
+      break;
+    }
+
+    case 4 - 1: {
+      static thread_local std::uniform_int_distribution<> date_diff_dist{0, 9};
+      const auto diff = date_diff_dist(random_engine);
+      const auto begin_date = calculate_date(boost::gregorian::date{1993, 01, 01}, diff);
+      const auto end_date = calculate_date(boost::gregorian::date{1993, 01, 01}, diff + 3);
```

Comment:
The standard wants dates between "first month of 1993 and the 10th month of **1997**." This seems to only generate dates up to 1993.10?

Reply:
Grr. I must have been past the Ballmer peak when I tried to transfer that into code...